Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -3,365 +3,2678 @@
+- when using location services you might want to check which [`CLAuthorizationStatus`](https://developer.apple.com/documentation/corelocation/clauthorizationstatus) is being returned by [`CLLocationManager.authorizationStatus()`](https://developer.apple.com/documentation/corelocation/cllocationmanager/1423523-authorizationstatus).
+- you could also verify the correct usage of the App Groups entitlement by hooking the [`containerURL(forSecurityApplicationGroupIdentifier:)`](https://developer.apple.com/documentation/foundation/filemanager/1412643-containerurl) method which returns the container directory associated with the specified security application group identifier.
+
+There is also a *visual* way to inspect the status of some app permissions when using the iPhone/iPad by opening "Settings" and scrolling down until you find the app you're interested in. When clicking on it, this will open the "ALLOW APP_NAME TO ACCESS" screen. However, not all permissions might be displayed yet. You will have to *trigger* them in order to be listed on that screen.
+
+![Settings Allow App Screen](Images/Chapters/0x06h/settings_allow_screen.png)
+ 
+For example, if you're testing a messaging app, try to share something. As you do it, surely the app will offer options to share photos or your location. If you click, for example, on share location, this will trigger the permission dialogue and the permission will be visible in "Settings" as well.
+
+
+### Testing for Sensitive Functionality Exposure Through IPC
+
+During implementation of a mobile application, developers may apply traditional techniques for IPC (such as using shared files or network sockets). The IPC system functionality offered by mobile application platforms should be used because it is much more mature than traditional techniques. Using IPC mechanisms with no security in mind may cause the application to leak or expose sensitive data.
+
+In contrast to Android's rich Inter-Process Communication (IPC) capability, iOS offers some rather limited options for communication between apps. In fact, there's no way for apps to communicate directly. In this section we will present the different types of indirect communication offered by iOS and how to test them. Here's an overview:
+
+- [Custom URL Schemes](#Custom-URL-Schemes)
+- [Universal Links](#Universal-Links)
+- [UIActivity Sharing](#UIActivity-Sharing)
+- [App Extensions](#App-Extensions)
+- [UIPasteboard](#UIPasteboard)
+
+#### Custom URL Schemes
+
+Please refer to the next section ["Testing Custom URL Schemes"](#Testing-Custom-URL-Schemes) for more information on what custom URL schemes are and how to test them.
+
+#### Universal Links
```

Comment:
Can you maybe elaborate quickly on the difference from Customer URL Schemes to  Universal Links?
Like, custom URL scheme for the facebook app would be fb://my/profile, and the Universal Link would be tied to the domain by configuring the app and hosting a AASA (apple-app-site-association).
This seems to explain it quite nicely https://medium.com/@abhimuralidharan/universal-links-in-ios-79c4ee038272

Reply:
This is now explained in the intro of "#### Universal Links", examples were added and the key benefits enhanced. I hope it is clearer now, please take a look :)